drivers/clock_control/clock_stm32_ll_u5.c

```diff
@@ -226,152 +197,76 @@ static void set_regu_voltage(uint32_t hclk_freq)
 	}
 }
 
-/*
- * Unconditionally switch the system clock source to HSI.
- */
 __unused
-static void clock_switch_to_hsi(uint32_t ahb_prescaler)
-{
-	/* Enable HSI if not enabled */
-	if (LL_RCC_HSI_IsReady() != 1) {
-		/* Enable HSI */
-		LL_RCC_HSI_Enable();
-		while (LL_RCC_HSI_IsReady() != 1) {
-		/* Wait for HSI ready */
-		}
-	}
-
-	/* Set HSI as SYSCLCK source */
-	LL_RCC_SetSysClkSource(LL_RCC_SYS_CLKSOURCE_HSI);
-	LL_RCC_SetAHBPrescaler(ahb_prescaler);
-	while (LL_RCC_GetSysClkSource() != LL_RCC_SYS_CLKSOURCE_STATUS_HSI) {
-	}
-}
-
-static void set_up_clk_msis(void)
+static int set_up_plls(void)
 {
-#if defined(STM32_SYSCLK_SRC_MSIS) || defined(STM32_PLL_SRC_MSIS)
-	/* Set MSIS Range */
-	LL_RCC_MSI_EnableRangeSelection();
+#if defined(STM32_PLL_ENABLED)
+	int r;
+	uint32_t vco_input_range;
 
-	LL_RCC_MSIS_SetRange(STM32_MSIS_RANGE << RCC_ICSCR1_MSISRANGE_Pos);
+	LL_RCC_PLL1_Disable();
 
-	if (IS_ENABLED(STM32_MSIS_PLL_MODE)) {
-		BUILD_ASSERT(STM32_LSE_ENABLED,
-			"MSI Hardware auto calibration needs LSE clock activation");
-		/* Enable MSI hardware auto calibration */
-		LL_RCC_MSI_EnablePLLMode();
+	/* Configure PLL source */
+	/* Can be HSE , HSI  MSI */
+	if (IS_ENABLED(STM32_PLL_SRC_HSE)) {
+		/* Main PLL configuration and activation */
+		LL_RCC_PLL1_SetMainSource(LL_RCC_PLL1SOURCE_HSE);
+	} else if (IS_ENABLED(STM32_PLL_SRC_MSIS)) {
+		/* Main PLL configuration and activation */
+		LL_RCC_PLL1_SetMainSource(LL_RCC_PLL1SOURCE_MSIS);
+	} else if (IS_ENABLED(STM32_PLL_SRC_HSI)) {
+		/* Main PLL configuration and activation */
+		LL_RCC_PLL1_SetMainSource(LL_RCC_PLL1SOURCE_HSI);
+	} else {
+		return -ENOTSUP;
 	}
 
-	/* Set MSIS Range */
-	LL_RCC_MSIS_Enable();
-
-	/* Wait till MSIS is ready */
-	while (LL_RCC_MSIS_IsReady() != 1) {
+	r = get_vco_input_range(STM32_PLL_M_DIVISOR, &vco_input_range);
```

Comment:
is it equivalent to `vco_input_range = get_vco_input_range(STM32_PLL_M_DIVISOR)`

Reply:
This way allows to generate errors on range issue.
For that part I'm fully reusing the existing code.